pkg/catalog/catalog.go

```diff
@@ -598,36 +598,25 @@ func (c *Catalog) ListRepositories(ctx context.Context, limit int, prefix, after
 	defer it.Close()
 	// seek for first item
 	afterRepositoryID := graveler.RepositoryID(after)
-	prefixRepositoryID := graveler.RepositoryID(prefix)
-	startPos := prefixRepositoryID
-	if afterRepositoryID > startPos {
-		startPos = afterRepositoryID
-	}
-	if startPos != "" {
-		it.SeekGE(startPos)
-	}
 
 	var repos []*Repository
 	for it.Next() {
 		record := it.Value()
-
-		if !strings.HasPrefix(string(record.RepositoryID), prefix) {
-			break
-		}
-
-		if record.RepositoryID == afterRepositoryID {
-			continue
-		}
-		repos = append(repos, &Repository{
-			Name:             record.RepositoryID.String(),
-			StorageNamespace: record.StorageNamespace.String(),
-			DefaultBranch:    record.DefaultBranchID.String(),
-			CreationDate:     record.CreationDate,
-			ReadOnly:         record.ReadOnly,
-		})
-		// collect limit +1 to return limit and has more
-		if len(repos) >= limit+1 {
-			break
+		if strings.Contains(string(record.RepositoryID), prefix) {
```

Comment:
This is a breaking change: it changes the meaning of the "prefix" parameter!  (It is also very confusing to users, who will probably not expect "prefix" to mean "substring").

Reply:
I guess I could change the name of the parameter- I don't see any other code that uses it

Reply:
Prefix is defined in our API and should still be supported, even if we wanted, we can't just change it.

Reply:
So we have to support prefix and also a search string?